src/uct/ib/dc/dc_mlx5.h

```diff
@@ -487,6 +533,9 @@ uct_dc_mlx5_iface_dci_has_outstanding(uct_dc_mlx5_iface_t *iface, int dci_index)
 {
     uct_rc_txqp_t *txqp;
 
+    if (!iface->tx.dcis[dci_index].initialized) {
```

Comment:
seems there are many places checking "initialized" but i would expect we will not get to them in the first place with an uninitialized dci - since an uninitialized dci should not be allocated by any endpoint

Reply:
But in random policy, any dci_index can be allocated and we need to track it somehow, another option would be to allocate all dcis ahead when random policy is used

Reply:
i see. so maybe reuse another field, such as qp number, to check if the dci is initialized?